elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -407,6 +407,7 @@ def compress_request(body, headers)
         end
 
         def decompress_response(body)
+          return body unless use_compression?
```

Comment:
Use compression shouldn't matter here. We should decompress the body if it's compressed for any reason.